### PR TITLE
Server and Client

### PR DESCRIPTION
**Pull request by Jurgenvs**
Functioning Server and Client! Needs some more tweaks and Client needs multithreading. Merge is already large enough as is.

### CODE REVIEW

**Comment by foresterre** [comment]
Currently this branch is merged with the pull requested branch 'CollectionScreen'.
However, that branch got refactored for a bit. Something to be careful about.

File: android/src/cg/group4/server/AndroidIDResolver.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package cg.group4.server;
+
+import android.content.Context;
+import android.provider.Settings;
+import cg.group4.client.UserIDResolver;
+
+public class AndroidIDResolver implements UserIDResolver {
+
+    protected Context cContext;
```
Comment (Surskit) [inline android/src/cg/group4/server/AndroidIDResolver.java]: Javadoc needed

File: core/src/cg/group4/Launcher.java [hunk 2]
```diff
@@ -73,10 +88,30 @@ public final void create() {
         cTimeKeeper = TimerStore.getInstance().getTimeKeeper();
 
         cStandUp = StandUp.getInstance();
+        
+        int fa = 0, fb = 0, fc = 0;
```
Comment (Surskit) [inline core/src/cg/group4/Launcher.java]: Temporary test code, do not forget to remove when all the features are attached.

File: core/src/cg/group4/game_logic/StandUp.java [hunk 3]
```diff
@@ -88,9 +92,14 @@ public void startStroll() {
...
      * @param cRewards rewards gained by the stroll.
      */
-    public void endStroll(final int cRewards) {
+    public void endStroll(final ArrayList<Integer> cRewards) {
         Gdx.app.log(TAG, "Ending stroll");
         cStroll = null;
+        
+        for (int score : cRewards) {
+        	Collectible c = cGenerator.generateCollectible(score);
+        	//ADD COLLECTIBLE TO COLLECTION HERE
```
Comment (Surskit) [inline core/src/cg/group4/game_logic/StandUp.java]: Add collectible to the storage

File: core/src/cg/group4/rewards/LocalCollectionReader.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    protected String cTag = this.getClass().getSimpleName();
+
+    /**
+     * Initialization of the local collection reader.
+     */
+    public LocalCollectionReader() {
+    }
+
+
```
Comment (Surskit) [inline core/src/cg/group4/rewards/LocalCollectionReader.java]: Unnecesary empty line (Yay for nitpicking!)

**Comment by Surskit** [comment]
Other than a couple of small things, everything seems in order.

**Comment by sirpolomotro** [comment]
To big a pull request... please keep them smaller....

File: core/src/cg/group4/client/query/Data.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package cg.group4.client.query;
+
+import java.io.Serializable;
+
+/**
+ * Data class that can be extended. Object extending this class can be send to the server.
+ */
+public abstract class Data implements Serializable { }
```
Comment (sirpolomotro) [inline core/src/cg/group4/client/query/Data.java]: Could be removed if you use Serializable instead of this class.. will refactor it later on...

**Pull request merged**